src/linear_algebra/eigenspace.lean

```diff
@@ -325,5 +325,21 @@ lemma generalized_eigenspace_eq_generalized_eigenspace_findim_of_le
   f.generalized_eigenspace μ k = f.generalized_eigenspace μ (findim K V) :=
 ker_pow_eq_ker_pow_findim_of_le hk
 
+/-- If `f` maps a subspace `p` into itself, then the generalized eigenspace of the restriction
+    of `f` to `p` is the part of the generalized eigenspace of `f` that lies in `p`. -/
+lemma generalized_eigenspace_restrict [field K] [vector_space K V]
+  (f : End K V) (p : submodule K V) (k : ℕ) (μ : K) (x : p) (hfp : ∀ (x : V), x ∈ p → f x ∈ p) :
+  generalized_eigenspace (linear_map.restrict f p hfp) μ k
+    = (f.generalized_eigenspace μ k).comap p.subtype :=
```

Comment:
Do you really like using the dot notation here? It's one of the case where it makes the whole expression reading backwards. `p.subtype` is pulling-back the submodule, not the other way around.

Reply:
I don't mind either way. I have removed the dot notation now.